component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -170,6 +201,43 @@ protected void processAuthenticationResponse(HttpServletRequest httpServletReque
         }
     }
 
+    private String getMatchedSubjectAttribute(String certAttributes, String subjectAttributePattern,
+            AuthenticationContext authenticationContext) throws AuthenticationFailedException {
+
+        LdapName ldapDN;
+        try {
+            ldapDN = new LdapName(certAttributes);
+        } catch (InvalidNameException e) {
+            throw new AuthenticationFailedException("error occurred while get the certificate claims", e);
+        }
+        String userNameAttribute = getAuthenticatorConfig().getParameterMap().get(X509CertificateConstants.USERNAME);
+        List<String> matchedStringList = new ArrayList<>();
+        for (Rdn distinguishNames : ldapDN.getRdns()) {
+            if (subjectPatternCompiled != null && userNameAttribute.equals(distinguishNames.getType())) {
+                Matcher m = subjectPatternCompiled.matcher(String.valueOf(distinguishNames.getValue()));
+                addMatchStringsToList(m, matchedStringList);
+            }
+        }
+        if (matchedStringList.isEmpty()) {
+            authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                    X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_NO_MATCHES_ERROR_CODE);
+            log.debug(X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_NO_MATCHES_ERROR);
+            throw new AuthenticationFailedException(X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_NO_MATCHES_ERROR);
+        } else if (matchedStringList.size() > 1) {
+            authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                    X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_MULTIPLE_MATCHES_ERROR_CODE);
+            log.debug("More than one value matched with the given regex, matches: " + Arrays.toString(matchedStringList.toArray()));
+            throw new AuthenticationFailedException("More than one value matched with the given regex");
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Setting X509Certificate username attribute: " + userNameAttribute + "and value is "
```

Comment:
log.debug("Setting X509Certificate username attribute: " + userNameAttribute + " , and value is "
                        + matchedStringList.get(0));

note the added space and comma

Reply:
done